nailgun/entities.py

```diff
@@ -1495,6 +1496,37 @@ def __init__(self, server_config=None, **kwargs):
         self._meta = {'api_path': 'api/v2/hostgroups', 'server_modes': ('sat')}
         super(HostGroup, self).__init__(server_config, **kwargs)
 
+    def read(self, entity=None, attrs=None, ignore=()):
+        if entity is None:
+            entity = type(self)(
+                self._server_config,
+                # pylint:disable=no-member
+                name=self.name,
+            )
```

Comment:
I think this entire block of code can be dropped. `entity` is not used anywhere else in this method, so there's no need to instantiate a `HostGroup` and assign it to `entity`. `super` will do that. Also, imagine the following use case:

```python
from nailgun import entities
host_group = entities.HostGroup(id=5).read()
```

After `HostGroup(id=1)` finishes executing, a `HostGroup` object will be instantiated, and that object will have just one public instance attribute set: `id`:

```python
>>> from nailgun import entities
>>> host_group = entities.HostGroup(id=5)
>>> host_group.get_values()
{'id': 5}
>>> vars(host_group).keys()
dict_keys(['_server_config', '_fields', '_meta', 'id'])
```

When `.read()` is executed, `self.name` is referenced. However, as you can see, no such public instance attribute exists.